### PR TITLE
Master reportalypse vin l10n fix

### PR DESCRIPTION
**Pull request by vin-odoo**

### CODE REVIEW

File: addons/l10n_eg/data/account_tax_report_data.xml [hunk 1]
```diff
@@ -21,13 +21,13 @@
                     <record id="tax_report_vat_return_sale_base_formula" model="account.report.expression">
                         <field name="label">balance</field>
                         <field name="engine">aggregation</field>
-                        <field name="formula">STD_SALE_B.balance + ZERO_SALE_B.balance + EXM_SALE_B.balance</field>
+                        <field name="formula">STDR_SALE_B.balance + ZEROR_SALE_B.balance + EXMP_SALE_B.balance</field>
```
Comment (oco-odoo) [inline addons/l10n_eg/data/account_tax_report_data.xml]: Why all these changes ?

File: addons/stock_account/tests/test_stockvaluationlayer.py [hunk 2]
```diff
@@ -986,26 +987,118 @@ def test_standard_auto_to_manual_2(self):
         self.assertEqual(len(self.product1.stock_valuation_layer_ids), 3)
 
 @tagged('post_install', '-at_install')
-class TestAngloSaxonAccounting(TestStockValuationCommon):
+class TestAngloSaxonAccounting(AccountTestInvoicingCommon):
```
Comment (oco-odoo) [inline addons/stock_account/tests/test_stockvaluationlayer.py]: What's that ? Why is it done so ? Does it have anything to do with this branch or is it a fix for a problem already present in master ?

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_eg/data/account_tax_report_data.xml (hunk 1 above)
Comment (vin-odoo) [reply addons/l10n_eg/data/account_tax_report_data.xml]: There is another report using the same line code, it causes issues when both are installed

File: addons/stock_account/tests/test_stockvaluationlayer.py (hunk 2 above)
Comment (vin-odoo) [reply addons/stock_account/tests/test_stockvaluationlayer.py]: It's running with the current company and thus is impacted by all things localisations.
IT's a fix for master problem, not reportalypse

File: addons/l10n_eg/data/account_tax_report_data.xml (hunk 1 above)
Comment (oco-odoo) [reply addons/l10n_eg/data/account_tax_report_data.xml]: Then let's prefix them both with country code

File: addons/stock_account/tests/test_stockvaluationlayer.py (hunk 2 above)
Comment (oco-odoo) [reply addons/stock_account/tests/test_stockvaluationlayer.py]: @Shawcker Could you remove this from this branch, create a new PR in master and have the stock team review it ?

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 3]
```diff
@@ -72,9 +72,21 @@ def _l10n_it_edi_is_self_invoice(self, invoice):
...
+        it_tax_report_id = self.env.ref('l10n_it.tax_report_vat').id
+        it_tax_report_expressions = self.env['account.report.expression'].search([('report_line_id.report_id', '=', it_tax_report_id)])
+        for expression in it_tax_report_expressions:
+            expression_xml_id = expression.get_external_id().get(expression.id, '')
+            if not expression_xml_id.startswith('l10n_it.tax_report_line_vj'):
+                continue
+
+            matching_tags_ids = expression._get_matching_tags().ids
+            if any(tag_id in invoice_lines_tag_ids for tag_id in matching_tags_ids):
+                return True
```
Comment (oco-odoo) [inline addons/l10n_it_edi/models/account_edi_format.py]: report_lines = self.env['account.report.line'].search([('report_id.country_id.code', '=', 'IT'), ('code', 'like', 'machin')])
vj_tags = report_lines.expression_ids._get_matching_tags()
return bool(vj_tags & invoice_lines_tag_ids)

Something like that :p 

@Shawcker

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 4]
```diff
@@ -215,8 +227,18 @@ def _l10n_it_goods_in_italy(self, invoice):
...
+        it_tax_report_expressions = self.env['account.report.expression'].search([('report_line_id.report_id', '=', it_tax_report_id)])
+        for expression in it_tax_report_expressions:
+            expression_xml_id = expression.get_external_id().get(expression.id, '')
+            if expression_xml_id != 'l10n_it.tax_report_line_vj3_tag':
+                continue
+
+            matching_tags_ids = expression._get_matching_tags().ids
+            if any(tag_id in invoice_lines_tag_ids for tag_id in matching_tags_ids):
+                return True
+        return False
```
Comment (oco-odoo) [inline addons/l10n_it_edi/models/account_edi_format.py]: idem

**Review by oco-odoo — CHANGES_REQUESTED** [review]

**Pull request merged**